### PR TITLE
Fix get snapshot 404 issue, and add API version validation

### PR DESCRIPTION
**Pull request by blueww**
Thanks for contribution! Please go through following checklist before sending PR.
1. Fix issue: https://github.com/Azure/Azurite/issues/293
2. Add API version validation , validate API version get from https://docs.microsoft.com/en-us/rest/api/storageservices/previous-azure-storage-service-versions

Don't have test now since:
1. For snapshot, JS test can't repro the issue. The issue is from PSH with XSCL
2. For API version, it's not easy to add test. We have manually test with PSH. Opened a task (P2) to add it later.

### PR Branch Destination

- For Azurite V3, please send PR to `dev` branch.
- For legacy Azurite V2, please send PR to `legacy-dev` branch.

### Always Add Test Cases

Make sure test cases are added to cover the code change.

### Development Guideline

Please go to CONTRIBUTION.md for steps about setting up development environment and recommended Visual Studio Code extensions.

### CODE REVIEW

File: src/blob/context/blobStorageContext.middleware.ts [hunk 1]
```diff
@@ -26,11 +28,14 @@ export default function blobStorageContextMiddleware(
 ): void {
   // Set server header in every Azurite response
   res.setHeader(HeaderConstants.SERVER, `Azurite-Blob/${VERSION}`);
+  const requestID = uuid();
+
+  const apiVersion = String(req.header("x-ms-version"));
```
Comment (XiaoningLiu) [inline src/blob/context/blobStorageContext.middleware.ts]: Should move "x-ms-version" into "constants.ts" `HeaderConstants`;

Should check the situation when `x-ms-version` is undefined, when we will fallback to Azurite default API Version.

File: src/common/utils/utils.ts [hunk 2]
```diff
@@ -1,8 +1,33 @@
...
   return date1 > date2 ? date2 : date1;
 }
+
+export function checkApiVersion(
+  inputApiVersion: string,
+  validApiVersions: Array<string>,
+  requestId: string
+): void {
+  if (
+    inputApiVersion !== undefined &&
```
Comment (XiaoningLiu) [inline src/common/utils/utils.ts]: parameter `inputApiVersion` is required in this definition. If you want to allow it tobe optional, you can update the type from `string` to `string | undefined`.

File: src/blob/context/blobStorageContext.middleware.ts (hunk 1 above)
Comment (XiaoningLiu) [reply src/blob/context/blobStorageContext.middleware.ts]:
```suggestion
  const apiVersion = req.header("x-ms-version");
  if (typeof apiVersion === "string") {
    checkApiVersion(apiVersion, ValidAPIVersion, requestID);
  }
```

File: src/blob/utils/constants.ts [hunk 3]
```diff
@@ -59,3 +59,24 @@ export const DEFAULT_BLOB_PERSISTENCE_ARRAY: StoreDestinationArray = [
     maxConcurrency: 10
   }
 ];
+
+export const ValidAPIVersion = [
```
Comment (XiaoningLiu) [inline src/blob/utils/constants.ts]:
```suggestion
export const ValidAPIVersions = [
```

File: src/queue/context/queueStorageContext.middleware.ts [hunk 4]
```diff
@@ -28,13 +30,17 @@ export default function queueStorageContextMiddleware(
   // Set server header in every Azurite response
   res.setHeader(HeaderConstants.SERVER, `Azurite-Queue/${VERSION}`);
 
+  const requestID = uuid();
+
+  const apiVersion = String(req.header("x-ms-version"));
```
Comment (XiaoningLiu) [inline src/queue/context/queueStorageContext.middleware.ts]: same comments with blob.

File: src/queue/utils/constants.ts [hunk 5]
```diff
@@ -88,3 +88,24 @@ export const DEFAULT_QUEUE_PERSISTENCE_ARRAY: StoreDestinationArray = [
     maxConcurrency: 1
   }
 ];
+
+export const ValidAPIVersion = [
```
Comment (XiaoningLiu) [inline src/queue/utils/constants.ts]:
```suggestion
export const ValidAPIVersions = [
```

File: src/blob/utils/constants.ts (hunk 3 above)
Comment (blueww) [reply src/blob/utils/constants.ts]: Fixed

File: src/queue/context/queueStorageContext.middleware.ts (hunk 4 above)
Comment (blueww) [reply src/queue/context/queueStorageContext.middleware.ts]: Has put "x-ms-version" to constants

File: src/blob/context/blobStorageContext.middleware.ts (hunk 1 above)
Comment (blueww) [reply src/blob/context/blobStorageContext.middleware.ts]: Has put "x-ms-version" to constants

Has check undefined `x-ms-version`,

File: src/common/utils/utils.ts (hunk 2 above)
Comment (blueww) [reply src/common/utils/utils.ts]: Has check undefined x-ms-version outside. and remove check for undefined inside.

File: src/queue/utils/constants.ts (hunk 5 above)
Comment (blueww) [reply src/queue/utils/constants.ts]: fixed

**Pull request merged**